bundles/binding/org.openhab.binding.girahomeserver/src/main/java/org/openhab/binding/girahomeserver/internal/GiraHomeServerBinding.java

```diff
@@ -0,0 +1,156 @@
+	 * method and must not be accessed anymore once the deactivate() method was called or before activate()
+	 * was called.
+	 */
+	private BundleContext bundleContext;
+
+	
+	/** 
+	 * the refresh interval which is used to poll values from the GiraHomeServer
+	 * server (optional, defaults to 60000ms)
+	 */
+	private long refreshInterval = 60000;
+	
+	
+	public GiraHomeServerBinding() {
+	}
+		
+	
+	/**
+	 * Called by the SCR to activate the component with its configuration read from CAS
+	 * 
+	 * @param bundleContext BundleContext of the Bundle that defines this component
+	 * @param configuration Configuration properties for this component obtained from the ConfigAdmin service
+	 */
+	public void activate(final BundleContext bundleContext, final Map<String, Object> configuration) {
+		this.bundleContext = bundleContext;
+
+		// the configuration is guaranteed not to be null, because the component definition has the
+		// configuration-policy set to require. If set to 'optional' then the configuration may be null
+		
+			
+		// to override the default refresh interval one has to add a 
+		// parameter to openhab.cfg like <bindingName>:refresh=<intervalInMs>
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString)) {
+			refreshInterval = Long.parseLong(refreshIntervalString);
+		}
+
+		// read further config parameters here ...
+
+		setProperlyConfigured(true);
+	}
+	
+	/**
+	 * Called by the SCR when the configuration of a binding has been changed through the ConfigAdmin service.
+	 * @param configuration Updated configuration properties
+	 */
+	public void modified(final Map<String, Object> configuration) {
+		// update the internal configuration accordingly
+	}
+	
+	/**
+	 * Called by the SCR to deactivate the component when either the configuration is removed or
+	 * mandatory references are no longer satisfied or the component has simply been stopped.
+	 * @param reason Reason code for the deactivation:<br>
+	 * <ul>
+	 * <li> 0 – Unspecified
+     * <li> 1 – The component was disabled
+     * <li> 2 – A reference became unsatisfied
+     * <li> 3 – A configuration was changed
+     * <li> 4 – A configuration was deleted
+     * <li> 5 – The component was disposed
+     * <li> 6 – The bundle was stopped
+     * </ul>
+	 */
+	public void deactivate(final int reason) {
+		this.bundleContext = null;
+		// deallocate resources here that are no longer needed and 
+		// should be reset when activating this binding again
+	}
+
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "GiraHomeServer Refresh Service";
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+		// the frequently executed code (polling) goes here ...
+		logger.debug("execute() method is called!");
```

Comment:
Implement AbstractBinding instead of AbstractActiveBinding if you are not going to do anything on the polling interval.